datafusion/core/src/datasource/physical_plan/parquet/page_filter.rs

```diff
@@ -548,7 +551,20 @@ impl<'a> PruningStatistics for PagesPruningStatistics<'a> {
     }
 
     fn row_counts(&self, _column: &datafusion_common::Column) -> Option<ArrayRef> {
-        None
+        // see https://github.com/apache/arrow-rs/blob/91f0b1771308609ca27db0fb1d2d49571b3980d8/parquet/src/file/metadata.rs#L979-L982
+        let mut first_row_index = self
+            .col_offset_indexes
```

Comment:
https://github.com/apache/arrow-rs/blob/91f0b1771308609ca27db0fb1d2d49571b3980d8/parquet/src/file/metadata.rs#L979-L982

Reply:
I think you can do this same calculation without allocating intermediate vec s-- something like this:

```rust
        let row_count_per_page = self
            .col_offset_indexes
            .windows(2)
            .map(|location| Some(location[1].first_row_index - location[0].first_row_index));

        Some(Arc::new(Int64Array::from_iter(row_count_per_page)))
```



🤔  the name `col_offset_indexes` is somewhat confusing to me as they are `PageLocation`s -- 

```
    col_offset_indexes: &'a Vec<PageLocation>,
```

maybe we could rename that field to `page_locations` 🤔

Reply:
In my mind `page_locations ` store the offsets in the column chunk ,  i think they are the same meaning 🤣